misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/SuspendedContext.java

```diff
@@ -112,21 +117,27 @@ public DebugSourceType getSourceType() {
     }
 
     public Path getBreakPointSourcePath() {
-        return Paths.get(breakPointSourcePath);
+        if (breakPointSourcePath == null) {
+            this.breakPointSourcePath = getSourcePath(frame);
+        }
+        return breakPointSourcePath;
     }
 
-    private String getSourcePath(StackFrameProxyImpl frame) {
+    private Path getSourcePath(StackFrameProxyImpl frame) {
```

Comment:
Since the return value can be `null`, shall we make the return value `Optional<Path>` ?

Reply:
+1. Changed in d1a92fb.